app/src/main/java/com/seafile/seadroid2/cameraupload/CameraUploadDBHelper.java

```diff
@@ -98,14 +138,90 @@ public void markAsUploaded(File file) {
         String path = file.getAbsolutePath();
         long modified = file.lastModified();
 
+        markAsUploaded(path, modified);
+    }
+
+    public void markAsUploaded(String path, long modified) {
+
         ContentValues values = new ContentValues();
         values.put(PHOTOCACHE_COLUMN_FILE, path);
         values.put(PHOTOCACHE_COLUMN_DATE_ADDED, modified);
 
         database.insert(PHOTOCACHE_TABLE_NAME, null, values);
     }
 
+    public void cleanCache() {
+        cleanPhotoCache();
+        cleanRepoCache();
+    }
+
     public void cleanPhotoCache() {
         database.delete(PHOTOCACHE_TABLE_NAME, null, null);
     }
+
+    public void cleanRepoCache() {
+        database.delete(REPOCACHE_TABLE_NAME, null, null);
+    }
+    public void saveRepoList(String bucketName, List<SeafDirent> list){
+        for (SeafDirent dirent : list) {
+            ContentValues values = new ContentValues();
+            values.put(REPOCACHE_COLUMN_BUCKET, bucketName);
+            values.put(REPOCACHE_COLUMN_FILE, dirent.name);
+            values.put(REPOCACHE_COLUMN_FILE_SIZE, dirent.size);
+            database.insert(REPOCACHE_TABLE_NAME, null, values);
+        }
+    }
+
+    public boolean isInRepo(File file){
+        String filename = file.getName();
+        long filelength = file.length();
+        String bucketName = file.getParent();
+        bucketName = bucketName.substring(bucketName.lastIndexOf("/")+1);
+        return isInRepo(bucketName, filename, filelength);
+    }
+
+    public boolean isInRepo(String bucketName, String filename, long filelength){
+        String prefix = filename.substring(0, filename.lastIndexOf("."));
+        String suffix = filename.substring(filename.lastIndexOf("."));
+//        Cursor c = database.query(
```

Comment:
?

Reply:
There are two search patterns. This search pattern is the same as the original code. Another we use in L196 is more strict than the original code, but faster.